File: LATPRAK4DPBO/C++/Vehicle.cpp
// Saya Marvel Ravindra Dioputra [2200481] LatPrak4 C++ dalam Mata Kuliah DPBO 
// untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

#pragma once
#include <iostream>
#include <string>

using namespace std;

class Vehicle {
private:
    string platNomor;       // atribut plat nomor 
    string merk;            // atribut merk
    string tahunProduksi;   // atribut tahun produksi
    string warna;           // atribut warna

public:
    // konstruktor
    Vehicle() {
        
    }

    Vehicle(string platNomor, string merk, string tahunProduksi, string warna) : platNomor(platNomor), merk(merk), tahunProduksi(tahunProduksi), warna(warna) {}

    // setter untuk atribut vehicle
    void setPlat(string platNomor) { 
        this->platNomor = platNomor; 
    }
    
    void setMerk(string merk) { 
        this->merk = merk; 
    }
    
    void setTahunProduksi(string tahunProduksi) { 
        this->tahunProduksi = tahunProduksi; 
    }
    
    void setWarna(string warna) { 
        this->warna = warna; 
    }

    // getter untuk atribut vehicle
    string getPlat() { 
        return platNomor; 
    }
    
    string getMerk() { 
        return merk; 
    }
    
    string getTahunProduksi() {
        return tahunProduksi;
    }

    string getWarna() {
        return warna;
    }

    // destruktor
    ~Vehicle() {

    }
};

File: LATPRAK4DPBO/python/vehicle.py
# Saya Marvel Ravindra Dioputra [2200481] LatPrak4 Python dalam Mata Kuliah DPBO 
# untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

# class vehicle
class Vehicle:
    # inisialisasi/instansiasi
    def __init__(self, plat_nomor, merk, tahun_produksi, warna):
        self.plat_nomor = plat_nomor            # atribut plat nomor
        self.merk = merk                        # atribut merk
        self.tahun_produksi = tahun_produksi    # atribut tahun produksi
        self.warna = warna                      # atribut warna

    # method menampilkan data vehicle
    def display(self):
        print("Plat Nomor:", self.plat_nomor)
        print("Merk:", self.merk)
        print("Tahun Produksi:", self.tahun_produksi)
        print("Warna:", self.warna)
